salt/utils/etcd_util.py

```diff
@@ -280,14 +345,23 @@ def update(self, fields, path=""):
         return keys
 
     def set(self, key, value, ttl=None, directory=False):
+        """
+        Write a file or directory, a higher interface to write
+        """
         return self.write(key, value, ttl=ttl, directory=directory)
 
     def write(self, key, value, ttl=None, directory=False):
+        """
+        Write a file or directory depending on directory flag
```

Comment:
I know this is existing code, and it might exist for a reason, but flags are usually horrible.

If I know enough to write:

```
write(..., directory=True)
...
write(..., directory=False)
```

Then I know enough to write:

```
write_directory(...)
...
write_file(...)
```

And also both options are shorter. :facepalm:

Reply:
I agree with this sentiment.  meant to check it was required for this interface, but am now realizing that this is in fact a utility module and probably not the case.  Might be good to create an issue and replace all uses with their corresponding method